src/inmanta/server/services/environment_metrics_service.py

```diff
@@ -0,0 +1,120 @@
+"""
+    Copyright 2022 Inmanta
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    Contact: code@inmanta.com
+"""
+import abc
+import logging
+from datetime import datetime, timedelta
+from enum import Enum
+from typing import List
+
+from inmanta.data import EnvironmentMetricsCounter, EnvironmentMetricsNonCounter
+from inmanta.server import SLICE_DATABASE, SLICE_ENVIRONMENT_METRICS, SLICE_TRANSPORT, protocol
+
+LOGGER = logging.getLogger(__name__)
+
+
+class MetricType(str, Enum):
+    count = "count"
+    non_count = "non_count"
+    compile_rate = "compile_rate"
```

Comment:
In python the convention is to use capital letter for the names of the venv instances.

Reply:
This is not respected at all in the codebase :p

Reply:
Indeed :p